wagtail_localize/views/edit_translation.py

```diff
@@ -1426,12 +1425,19 @@ def machine_translate(request, translation_id):
                             "data": translations[string].data,
                             "translation_type": StringTranslation.TRANSLATION_TYPE_MACHINE,
                             "tool_name": translator.display_name,
-                            "last_translated_by": request.user,
+                            "last_translated_by": user,
                             "has_error": False,
                             "field_error": "",
                         },
                     )
+        return True
+    return False
 
+
+@require_POST
+def machine_translate(request, translation_id):
+    if apply_machine_translation(translation_id, request.user):
```

Comment:
`apply_machine_translation` is doing too many things and can have unintended consequences on the sync translation form.

I would rather it purely takes the translation id, user, and the machine translator instance and it *applies* the machine translation.

Move the additional checks in the view (and ditto, as additional checks on the sync translation form init and validation)